src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2358 @@ class GDIFileStream : public InputStream
+        }
+
+        file->Close();
+    }
+}
+
+bool WorldHeightMap::Parse_Object_Data_Chunk(DataChunkInput &file, DataChunkInfo *info, void *user_data)
+{
+    return ((WorldHeightMap *)file.m_userData)->Parse_Object_Data(file, info, user_data, info->version >= 2);
+}
+
+bool WorldHeightMap::Parse_Object_Data(DataChunkInput &file, DataChunkInfo *info, void *user_data, bool read_dict)
+{
+    MapObject *pPrevious = (MapObject *)file.m_currentObject;
+    Coord3D loc;
+    loc.x = file.Read_Real32();
+    loc.y = file.Read_Real32();
+    loc.z = file.Read_Real32();
+    float f1 = -1000.0f;
+    float f2 = 1593.75f;
+
+    if (info->version <= 2) {
+        loc.z = 0.0f;
+    }
+
+    float angle = file.Read_Real32();
+    int flags = file.Read_Int32();
+    Utf8String name = file.Read_AsciiString();
+    Dict d(0);
+
+    if (read_dict) {
+        d = file.Read_Dict();
+    }
+
+    if (loc.z >= f1 && loc.z <= f2) {
+        MapObject *obj = NEW_POOL_OBJ(MapObject, loc, name, angle, flags, &d, g_theThingFactory->Find_Template(name, false));
+
+        if (obj->Get_Properties()->Get_Type(g_waypointIDKey) == Dict::DICT_INT) {
+            obj->Set_Is_Waypoint();
+        }
+
+        if (obj->Get_Properties()->Get_Type(g_lightHeightAboveTerrainKey) == Dict::DICT_REAL) {
+            obj->Set_Is_Light();
+        }
+
+        if (obj->Get_Properties()->Get_Type(g_scorchTypeKey) == Dict::DICT_INT) {
+            obj->Set_Is_Scorch();
+        }
+
+        if (pPrevious) {
+            captainslog_dbgassert(MapObject::s_theMapObjectListPtr && !pPrevious->Get_Next(), "Bad linkage.");
+
+            pPrevious->Set_Next_Map(obj);
+        } else {
+            captainslog_dbgassert(!MapObject::s_theMapObjectListPtr, "Bad linkage.");
+            MapObject::s_theMapObjectListPtr = obj;
+        }
+
+        file.m_currentObject = obj;
+    } else {
+        captainslog_debug("Removing object at z height %f", loc.z);
+    }
+
+    return true;
+}
+
+struct TTargaHeader
+{
+    unsigned char id_length;
+    unsigned char color_map_type;
+    unsigned char image_type;
+    unsigned char color_map_info[5];
+    short x_origin;
+    short y_origin;
+    short image_width;
+    short image_height;
+    unsigned char pixel_depth;
+    unsigned char flags;
+};
+
+int WorldHeightMap::Count_Tiles(InputStream *str, bool *b)
+{
+    TTargaHeader hdr;
+
+    if (b) {
+        *b = false;
+    }
+
+    if (str->Read(&hdr, sizeof(hdr)) != sizeof(hdr)) {
+        return 0;
+    }
+
+    int w = hdr.image_width / 64;
+    int h = hdr.image_height / 64;
+
+    if (hdr.color_map_type) {
+        return 0;
+    }
+
+    if (hdr.image_type != 2 && hdr.image_type != 10) {
+        return 0;
+    }
+
+    if (hdr.pixel_depth < 24u) {
+        return 0;
+    }
+
+    if (hdr.pixel_depth > 32u) {
+        return 0;
+    }
+
+    if (w > 10 || h > 10) {
+        return 0;
+    }
+
+    if (w >= 10 && h >= 10) {
+        return 100;

```

Comment:
these should all be `10*10` or `9*9` or ... to make it clear that its just `w*h`. In fact i would just get rid of this comletly and do:
```
if(w >0 && h>0){
   const auto smallest = std::min(w,h);
   return smallest * smallest;
}

Reply:
So BFME2 has a replacement for this actually
```
    for ( int i = 10; i > 0; --i )
    {
        if ( w >= i && h >= i )
        {
            return i * i;
        }
    }
```
looks like it would do the exact same thing but without the ugly

Reply:
fixed with the BFME2 change.